pmm-app/src/pmm-dbaas/components/XtraDB/XtraDB.hooks.ts

```diff
@@ -0,0 +1,44 @@
+import { useState, useEffect } from 'react';
+import { processPromiseResults, filterFulfilled, FulfilledPromiseResult } from 'shared/components/helpers/promises';
+import { DATABASE_LABELS } from 'shared/core';
+import { Kubernetes } from '../Kubernetes/Kubernetes.types';
+import { XtraDBCluster, GetXtraDBClustersAction } from './XtraDB.types';
+import { XtraDBService, toModel } from './XtraDB.service';
+
+export const useXtraDBClusters = (kubernetes: Kubernetes[]): [
+  XtraDBCluster[],
+  GetXtraDBClustersAction,
+  boolean
+] => {
+  const [xtraDBClusters, setXtraDBClusters] = useState<XtraDBCluster[]>([]);
+  const [loading, setLoading] = useState(false);
```

Comment:
I suggest to use something like `pendingRequest` instead of `loading`, in accordance to what we're doing for new code